odoo/fields.py

```diff
@@ -1466,10 +1465,22 @@ def determine_domain(self, records, operator, value):
 class Boolean(Field):
     """ Encapsulates a :class:`bool`. """
     type = 'boolean'
-    column_type = ('bool', 'bool')
 
-    def convert_to_column(self, value, record, values=None, validate=True):
-        return bool(value)
+    @property
+    def column_type(self):
+        return ('jsonb', 'jsonb') if self.company_dependent else ('bool', 'bool')
```

Comment:
Given the implementation, I understand why this is needed, however I find it ugly and breaking abstractions because if you have a new field type, you need to do this.
You could define `column_type` as a property on Field as `jsonb if self.company_dependent else self._column_type`.